### PR TITLE
chore: upgrade mockative to 2.2.0 (ACOL-150)

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Mockative was updated to v2
We gotta keep up-to-date if we want to ever update to Kotlin 2.0.0 and KSP 2. Mockative most likely won't bring support to 1.x to these versions.

### Solutions

I don't even know where to start.

ThatsHugeEnormousGIF

For an overview of the new syntax, [see this](https://github.com/mockative/mockative/blob/main/wiki/MIGRATING-TO-2.md).

TL;DR:

```kotlin
// Mockative 1
given(api)
    .suspendFunction(api::fetch)
    .whenInvokedWith(any())
    .thenReturn(response)

// Mockative 2
coEvery { api.fetch(any()) }
    .returns(response)    
```

or:

```kotlin
// Mockative 1
given(mapper)
    .invocation { transform(response) }
    .thenReturn(dto)

// Mockative 2
every { mapper.transform(response) }
    .returns(dto)
```

----

What else does it imply for our project?
- `runBlocking` was added into Arrangements `init` block to be able to configure it, which shouldn't be an issue.
- Our `Arrangement` functions which accept `Matcher` arguments have become harder to use, needing to call with `EqualsMatcher` instead of `eq()`, or similar. To be fair, I think the whole `Arrangement` now is _almost_ not worth doing any more, given how simple it became to write directly into Mockative

#### Test Coverage

Yes. That's pretty much the whole PR.

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by yamilmedina — APPROVED** [review]
Cool effort and hustle with this big update 👏 
I'll just leave it approved and trust the all mighty CI.

**Review by MohamadJaara — COMMENTED** [review]
LGTM

**Review by ohassine — APPROVED** [review]

**Comment by vitorhugods** [comment]
It seems Android doesn't like the mocking in Mockative v2. This works on normal JVM, not on Android.
I'm looking into solutions

```
java.lang.NullPointerException: Attempt to read from null array
at javassist.util.proxy.SecurityActions.getCallerClass(SecurityActions.java:54)
at javassist.util.proxy.DefineClassHelper$Java7.defineClass(DefineClassHelper.java:174)
at javassist.util.proxy.DefineClassHelper.toClass(DefineClassHelper.java:260)
at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:154)
at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:640)
at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:624)
at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:560)
at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:481)
at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:828)
at io.mockative.fake.ValueOfJVM.makeValueOf(ValueOf.kt:55)
at io.mockative.fake.ValueOfJVM.makeValueOf(ValueOf.kt:31)
at io.mockative.fake.ValueOfKt.valueOf(ValueOf.kt:84)
at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invokeSuspend(LoginRepositoryTest.kt:134)
at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invoke(Unknown Source:8)
at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invoke(Unknown Source:2)
at io.mockative.Mockable$Companion.record(Mockable.kt:358)
```

**Comment by yamilmedina** [comment]
> It seems Android doesn't like the mocking in Mockative v2. This works on normal JVM, not on Android. I'm looking into solutions
> 
> ```
> java.lang.NullPointerException: Attempt to read from null array
> at javassist.util.proxy.SecurityActions.getCallerClass(SecurityActions.java:54)
> at javassist.util.proxy.DefineClassHelper$Java7.defineClass(DefineClassHelper.java:174)
> at javassist.util.proxy.DefineClassHelper.toClass(DefineClassHelper.java:260)
> at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:154)
> at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:640)
> at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:624)
> at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:560)
> at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:481)
> at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:828)
> at io.mockative.fake.ValueOfJVM.makeValueOf(ValueOf.kt:55)
> at io.mockative.fake.ValueOfJVM.makeValueOf(ValueOf.kt:31)
> at io.mockative.fake.ValueOfKt.valueOf(ValueOf.kt:84)
> at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invokeSuspend(LoginRepositoryTest.kt:134)
> at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invoke(Unknown Source:8)
> at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invoke(Unknown Source:2)
> at io.mockative.Mockable$Companion.record(Mockable.kt:358)
> ```

If this is the only test failing, it might be an option to ignore it, so this massive work/change does not get lost and conflicted later.
